src/forcedRetargeting.ts

```diff
@@ -0,0 +1,120 @@
+// Code came from https://github.com/spring-media/react-shadow-dom-retarget-events/blob/516dafb756d8e3daaadaf9f8b48f85c6811e93e7/index.js
+/* eslint-disable no-param-reassign */
+/* eslint-disable  @typescript-eslint/explicit-function-return-type */
+const reactEvents = [
+  'onMouseDown',
+  'onMouseMove',
+  'onMouseOut',
+  'onMouseOver',
+  'onMouseUp',
+];
+
+function findReactProperty(item: any, propertyPrefix: string): any {
+  // eslint-disable-next-line no-restricted-syntax
+  for (const key in item) {
+    // eslint-disable-next-line no-prototype-builtins
+    if (item.hasOwnProperty(key) && key.includes(propertyPrefix)) {
+      return item[key];
+    }
+  }
+  return null;
+}
+
+function findReactEventHandlers(item: any): any {
+  return findReactProperty(item, '__reactEventHandlers');
+}
+
+function findReactComponent(item: any): any {
+  return findReactProperty(item, '_reactInternal');
+}
+
+function findReactProps(component: any): any {
+  if (!component) return undefined;
+  if (component.memoizedProps) return component.memoizedProps; // React 16 Fiber
+  if (component._currentElement && component._currentElement.props) return component._currentElement.props; // React <=15
+  return null;
+}
+
+function dispatchEvent(
+  event: any,
+  eventType: string,
+  componentProps: any,
+): any {
+  event.persist = function() {
+    event.isPersistent = function() {
+      return true;
+    };
+  };
+
+  if (componentProps[eventType]) {
+    componentProps[eventType](event);
+  }
+}
+
+function composedPath(
+  el: HTMLElement | null,
+): (Node | (Window & typeof globalThis))[] | null {
+  const path = [];
+  while (el) {
+    path.push(el);
+    if (el.tagName === 'HTML') {
+      path.push(document);
+      path.push(window);
+      return path;
+    }
+    el = el.parentElement;
+  }
+  return null;
+}
+
+export default function retargetEvents(shadowRoot: Node): () => void {
+  const removeEventListeners: (() => void)[] = [];
+
+  reactEvents.forEach((reactEventName) => {
+    const nativeEventName = reactEventName.replace(/^on/, '').toLowerCase();
+
+    function retargetEventListener(event: Event | any): void {
+      const path = event.path
+        || (event.composedPath && event.composedPath())
+        || composedPath(event.target);
+
+      for (let i = 0; i < path.length; i += 1) {
```

Comment:
You'll get an error here I think if the `composedPath` method returns null. We might want to have an escape hatch like `(path || []).length`.

Reply:
The better answer seemed to just return an empty array in composed path. That does tell me that their code would have broken in that case anyway